### PR TITLE
fix: fill missing clusterIP for service in egctl x translate

### LINKED ISSUE #3707: egctl translate doesn't work if clusterIP is not present in for k8s Service
*Description*:
> When using egctl to translate from k8s config to xDS config, if a clusterIP is not mentioned in Service, then translation fails but if I add a clusterIP the translation succeeds. To me it looks like a bug because clusterIP is not a requirement in k8s config and i am not sure why it should be required here.

*Repro steps*:
```bash
cat <<EOF | egctl x translate --from gateway-api --to xds -f -
apiVersion: gateway.networking.k8s.io/v1
kind: GatewayClass
metadata:
  name: eg
spec:
  controllerName: gateway.envoyproxy.io/gatewayclass-controller
---
apiVersion: gateway.networking.k8s.io/v1
kind: Gateway
metadata:
  name: eg
spec:
  gatewayClassName: eg
  listeners:
    - name: http
      protocol: HTTP
      port: 80
---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: backend
---
apiVersion: v1
kind: Service
metadata:
  name: backend
  labels:
    app: backend1
    service: backend1
spec:
  ports:
    - name: http
      port: 8080
      targetPort: 8080
  selector:
    app: backend
---
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: routes
spec:
  parentRefs:
    - name: eg
  rules:
    - backendRefs:
        - group: ""
          kind: Service
          name: backend
          port: 8080
      matches:
        - path:
            type: PathPrefix
            value: /service
EOF
```
fails with below error
```bash
Error: failed to translate xds ir for key envoy-gateway-system/eg value &{AccessLog:0x14000694730 Tracing:<nil> Metrics:<nil> HTTP:[0x140006bc000] TCP:[] UDP:[] EnvoyPatchPolicies:[] FilterOrder:[]}, error:validation failed for xds resource cluster_name:"httproute/envoy-gateway-system/routes/rule/0" endpoints:{locality:{region:"httproute/envoy-gateway-system/routes/rule/0/backend/0"} lb_endpoints:{endpoint:{address:{socket_address:{port_value:8080}}} load_balancing_weight:{value:1}} load_balancing_weight:{value:1}}, err: invalid ClusterLoadAssignment.Endpoints[0]: embedded message failed validation | caused by: invalid LocalityLbEndpoints.LbEndpoints[0]: embedded message failed validation | caused by: invalid LbEndpoint.Endpoint: embedded message failed validation | caused by: invalid Endpoint.Address: embedded message failed validation | caused by: invalid Address.SocketAddress: embedded message failed validation | caused by: invalid SocketAddress.Address: value length must be at least 1 runes
failed to translate xds ir for key envoy-gateway-system/eg value &{AccessLog:0x14000694730 Tracing:<nil> Metrics:<nil> HTTP:[0x140006bc000] TCP:[] UDP:[] EnvoyPatchPolicies:[] FilterOrder:[]}, error:validation failed for xds resource cluster_name:"httproute/envoy-gateway-system/routes/rule/0" endpoints:{locality:{region:"httproute/envoy-gateway-system/routes/rule/0/backend/0"} lb_endpoints:{endpoint:{address:{socket_address:{port_value:8080}}} load_balancing_weight:{value:1}} load_balancing_weight:{value:1}}, err: invalid ClusterLoadAssignment.Endpoints[0]: embedded message failed validation | caused by: invalid LocalityLbEndpoints.LbEndpoints[0]: embedded message failed validation | caused by: invalid LbEndpoint.Endpoint: embedded message failed validation | caused by: invalid Endpoint.Address: embedded message failed validation | caused by: invalid Address.SocketAddress: embedded message failed validation | caused by: invalid SocketAddress.Address: value length must be at least 1 runes
```

Following command works
```bash
cat <<EOF | egctl x translate --from gateway-api --to xds -f -
apiVersion: gateway.networking.k8s.io/v1
kind: GatewayClass
metadata:
  name: eg
spec:
  controllerName: gateway.envoyproxy.io/gatewayclass-controller
---
apiVersion: gateway.networking.k8s.io/v1
kind: Gateway
metadata:
  name: eg
spec:
  gatewayClassName: eg
  listeners:
    - name: http
      protocol: HTTP
      port: 80
---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: backend
---
apiVersion: v1
kind: Service
metadata:
  name: backend
  labels:
    app: backend1
    service: backend1
spec:
  clusterIP: 10.0.171.239
  ports:
    - name: http
      port: 8080
      targetPort: 8080
  selector:
    app: backend
---
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: routes
spec:
  parentRefs:
    - name: eg
  rules:
    - backendRefs:
        - group: ""
          kind: Service
          name: backend
          port: 8080
      matches:
        - path:
            type: PathPrefix
            value: /service
EOF
```

### PR DESCRIPTION
**Pull request by shawnh2**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #3707

### CODE REVIEW

File: internal/cmd/egctl/translate.go [hunk 1]
```diff
@@ -792,6 +793,10 @@ func kubernetesYAMLToResources(str string, addMissingResources bool) (*gatewayap
 				},
 				Spec: typedSpec.(corev1.ServiceSpec),
 			}
+			// fill with dummy IP when service clusterIP is empty
+			if len(service.Spec.ClusterIP) == 0 {
```
Comment (arkodg) [inline internal/cmd/egctl/translate.go:797]: imo we should only do this if `--add-missing-resources` is set

Comment (shawnh2) [reply internal/cmd/egctl/translate.go]: sounds right, added

**Review by arkodg — DISMISSED** [review]
LGTM thanks !

**Review by Xunzhuo — APPROVED** [review]

**Review by cnvergence — APPROVED** [review]

**Pull request merged**